local ReplicatedStorage = game:GetService("ReplicatedStorage")
local jecs = require(ReplicatedStorage.Modules.Imports.jecs)
local world = require(script.Parent.jecs_world)

type Entity<T = nil> = jecs.Entity<T>

local components: {
	Character: Entity<Model>,
	Mob: Entity<Model>,
	Model: Entity<Model>,
	Player: Entity<Player>,
	Target: Entity<Model>,
	Transform: Entity<{ new: CFrame, old: CFrame }>,
	Velocity: Entity<Vector3>,
	Previous: Entity<any>,
    Health: Entity<{current: number, new: number, old: number, tick: number, tickspeed: number}>,
    Energy: Entity<{current: number, max: number, new: number, old: number, tick: number, tickspeed: number}>,
    Attacking: Entity<{value: boolean, duration: number}>,
    Stun: Entity<{value: boolean, duration: number}>,
    NoJump: Entity<{value: boolean, duration: number}>,
    ToSpeed: Entity<number>,
    NoRotate: Entity<{value: boolean, duration: number}>,
    Gripping: Entity<{target: Model, value: boolean}>,
    Sliding: Entity<{direction: CFrame, value: boolean}>,
    WallRunning: Entity<boolean>,
    Dashing: Entity<boolean>,
    Combat: Entity<{ equipped: boolean, animation: string, weapon: string }>,
    Blocking: Entity<{value: boolean, duration: number}>,
    Carrying: Entity<{target: Model, value: boolean}>,
    BeingGripped: Entity<boolean>,
    BeingCarried: Entity<boolean>,
    DeathLocation: Entity<CFrame>,
    Knocked: Entity<{duration: number, value: boolean}>,
    IFrame: Entity<{duration: number, value: boolean}>,
    Sprinting: Entity<{value: boolean}>,
    Ragdoll: Entity<{duration: number, value: boolean}>,
    CantMove: Entity<{value: boolean, duration: number}>,
    Dead: Entity<boolean>,
    Weapon: Entity<{name: string, type: string}>,
    Swing: Entity<any>,
    KeysLogged: Entity<number>,
    Light: Entity<{value: boolean, duration: number}>,
    NoDash: Entity<{value: boolean, duration: number}>,
    LastHit: Entity<number>,
    AttackInfo: Entity<{name: string, type: string, damage: number, damagetype: number, scales: boolean}>,
    CritCD: Entity<{value: boolean, duration: number}>,
    Heavy: Entity<{value: boolean, duration: number}>,
    IgnoreParry: Entity<{value: boolean, duration: number}>,
    Phase: Entity<RBXScriptSignal>,
    System: Entity<{callback: (any) -> (), name: string}>,
    Damage: Entity<{amount: number, victim: Model, type: string}>,
    NoHurt: Entity<{value: boolean, duration: number}>,
    ParryTick: Entity<{value: boolean, duration: number}>,
    QDC: Entity<{value: boolean, duration: number}>,
    BBRegen: Entity<{value: boolean, duration: number}>,
    LastMove: Entity<{move: string, tim: number}>,
    LastAction: Entity<{action: string, tim: number}>,
    ParryStun: Entity<{value: boolean, duration: number}>,
    InAir: Entity<boolean>,
    Utility: Entity<{value: boolean, duration: number}>,
    Transmutable: Entity<boolean>,
    Victim: Entity<Model>,
    Action: Entity<{value: boolean, duration: number}>,
    BlockBar: Entity<{Value: number, MaxValue: number}>,
    BlockBroken: Entity<boolean>,
    NoRagdoll: Entity<boolean>,
    InCombat: Entity<{duration: number, value: boolean}>,
    CustomOst: Entity<boolean>,
    Locked: Entity<{value: boolean, duration: number}>,
    TWall: Entity<boolean>,
    DependsOn: Entity<any>,
    Event: Entity<RBXScriptSignal>,
    Name: Entity<string>,
    After: Entity<any>,
    Special: Entity<boolean>,
    Attack: Entity<{type: string, name: string}>,
    Dialogue: Entity<{npc: Model, name: string, inrange: boolean, state: string}>,
    ActiveQuest: Entity<{npcName: string, questName: string, startTime: number, progress: {}}>,
    CompletedQuest: Entity<{npcName: string, questName: string, completedTime: number}>,
    QuestAccepted: Entity<{npcName: string, questName: string, acceptedAt: number}>,
    QuestData: Entity<{Description: string, Rewards: {}}>,
    QuestHolder: Entity<boolean>,
    Passive: Entity<{name: string, description: string?,cooldown : boolean}>,
    PassiveHolder: Entity<table>,
    Inventory: Entity<{items: {[number]: {name: string, typ: string, quantity: number, singleuse: boolean, slot: number?}}, maxSlots: number}>,
    Item: Entity<{name: string, typ: string, quantity: number, singleuse: boolean, description: string?, icon: string?, stackable: boolean?}>,
    Requirement: Entity<{type: string, value: any, description: string?}>,
    Hotbar: Entity<{slots: {[number]: number?}, activeSlot: number?}>,
    InventoryChanged: Entity<number>,
    ClockTime: Entity<{timeOfDay: number, dayLength: number, startTime: number}>,
} =
	{
		Character = world:component(),
		Mob = world:component(),
		Model = world:component(),
		Player = world:component(),
		Target = world:component(),
		Transform = world:component(),
		Velocity = world:component(),
		Previous = world:component(),
        Health = world:component(),
        Energy = world:component(),
        Attacking = world:component(),
        Stun = world:component(),
        NoRotate = world:component(),
        Gripping = world:component(),
        Sliding = world:component(),
        WallRunning = world:component(),
        Dashing = world:component(),
        Combat = world:component(),
        Blocking = world:component(),
        Carrying = world:component(),
        BeingGripped = world:component(),
        BeingCarried = world:component(),
        DeathLocation = world:component(),
        Knocked = world:component(),
        IFrame = world:component(),
        Sprinting = world:component(),
        Ragdoll = world:component(),
        CantMove = world:component(),
        Dead = world:component(),
        Weapon = world:component(),
        Swing = world:component(),
        NoJump = world:component(),
        ToSpeed = world:component(),
        KeysLogged = world:component(),
        Light = world:component(),
        NoDash = world:component(),
        LastHit = world:component(),
        AttackInfo = world:component(),
        CritCD = world:component(),
        Heavy = world:component(),
        IgnoreParry = world:component(),
        System = world:component(),
        Damage = world:component(),
        NoHurt = world:component(),
        ParryTick = world:component(),
        ParryStun = world:component(),
        InAir = world:component(),
        Utility = world:component(),
        Victim = world:component(),
        Action = world:component(),
        BlockBar = world:component(),
        BBRegen = world:component(),
        BlockBroken = world:component(),
        Armor = world:component(),
        QDC = world:component(),
        NoRagdoll = world:component(),
        InCombat = world:component(),
        CustomOst = world:component(),
        Locked = world:component(),
        TWall = world:component(),
        Phase = world:component(),
        DependsOn = world:component(),
        Event = world:component(),
        Name = world:component(),
        After = world:component(),
        Special = world:component(),
        Attack = world:component(),
        Dialogue = world:component(),
        ActiveQuest = world:component(),
        CompletedQuest = world:component(),
        QuestAccepted = world:component(),
        QuestData = world:component(),
        QuestHolder = world:component(),
        Passive = world:component(),
        PassiveHolder = world:component(),
        Inventory = world:component(),
        Item = world:component(),
        Requirement = world:component(),
        Hotbar = world:component(),
        InventoryChanged = world:component(),
        ClockTime = world:component(),
	}

    for name, component in components :: {[string]: jecs.Entity} do 
        -- print("Setting component:", name, component)
        if component == nil then
            -- error("Component is nil: " .. name)
        end
        world:set(component, jecs.Name, name)
        -- print("Component set successfully:", name)
    end

return table.freeze(components)